dask/array/core.py

```diff
@@ -5745,6 +5749,8 @@ def _subset_to_indexed_axes(iterable):
         sorted_keys = keys.flat[sortidx]  # flattens
         sorted_inblock_idxs = [_.flat[sortidx] for _ in inblock_idxs]
         sorted_outblock_idx = outblock_idx.flat[sortidx]
+        dtype = np.min_scalar_type(max(sorted_outblock_idx))
+        sorted_outblock_idx = sorted_outblock_idx.astype(dtype)
```

Comment:
We already know this:
```suggestion
        dtype = np.min_scalar_type(max_chunk_point_dimensions)
        sorted_outblock_idx = sorted_outblock_idx.astype(dtype, copy=False)
```

Reply:
or see comment below

Reply:
done